x-pack/legacy/plugins/ml/public/services/ml_server_info.ts

```diff
@@ -43,3 +55,24 @@ export function newJobDefaults(): MlServerDefaults {
 export function newJobLimits(): MlServerLimits {
   return limits;
 }
+
+export function cloudId(): string | null {
+  return cloudInfo.cloudId;
+}
+
+export function isCloud(): boolean {
+  return cloudInfo.isCloud;
+}
+
+export function cloudDeploymentId(): string | null {
+  if (cloudInfo.cloudId === null) {
+    return null;
+  }
+  const tempCloudId = cloudInfo.cloudId.replace(/^.+:/, '');
+  try {
+    const matches = atob(tempCloudId).match(/(?<=\$).+(?=\$)/);
+    return matches !== null ? matches[0] : null;
```

Comment:
nit: you can use [test](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test)